### PR TITLE
swoop caboose minio client

### PR DESCRIPTION
**Pull request by hectormachin**

### CODE REVIEW

File: pkg/s3/s3driver.go [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+}
+
+func (s3 *S3Driver) AddFlags(fs *pflag.FlagSet) {
+	fs.StringVar(
+		&s3.Endpoint,
+		"s3-endpoint",
+		"",
+		"swoop s3 endpoint (required; SWOOP_S3_ENDPOINT)",
+	)
+	cobra.MarkFlagRequired(fs, "s3-endpoint")
```
Comment (jkeifer) [inline pkg/s3/s3driver.go]: I don't think this should be required.

File: pkg/s3/s3driver.go [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+		"swoop s3 access key (required; SWOOP_S3_ACCESS_KEY)",
+	)
+	cobra.MarkFlagRequired(fs, "s3-access-key")
+	fs.StringVar(
+		&s3.SecretKey,
+		"s3-secret-key",
+		"",
+		"swoop s3 secret key (required; SWOOP_S3_SECRET_KEY)",
+	)
+	cobra.MarkFlagRequired(fs, "s3-role-arn")
```
Comment (jkeifer) [inline pkg/s3/s3driver.go]: I am uncertain if these should be required. Is it possible to get S3 access without them, like when running in AWS with a role that provides it?

Comment (hectormachin) [reply pkg/s3/s3driver.go]: I think it makes sense to remove the required flag condition here.

File: pkg/s3/s3driver.go (hunk 1 above)
Comment (hectormachin) [reply pkg/s3/s3driver.go]: Agreed, removed.

**Review by jkeifer — APPROVED** [review]

**Pull request merged**